[Bug]: HotRod run failed if using the old env variable

### What happened?

The HotRod can fail if we try to run it in Kubernetes with Jaeger Agent (with a Service name `jaeger-agent`) or with one of the old parameters like `JAEGER_AGENT_PORT` (that have `OTEL_EXPORTER_*` alternatives).

During a short investigation, I found the issue added after PR https://github.com/jaegertracing/jaeger/pull/5437.
After removing `expvar` the method 

```go
jaegerclientenv2otel.MapJaegerToOtelEnvVars(logger)
```
here https://github.com/jaegertracing/jaeger/blob/main/examples/hotrod/cmd/root.go#L59 
now calling with a non-initialized `logger` variable. 

As a result, the logger in `jaegerclientenv2otel.MapJaegerToOtelEnvVars` will fail
https://github.com/jaegertracing/jaeger/blob/main/internal/jaegerclientenv2otel/envvars.go#L57
with nil pointer.

### Steps to reproduce

1. Add in ENV the variable like `JAEGER_AGENT_HOST=<value>` or `JAEGER_AGENT_PORT=<value>` (in Kubernetes deployment or for local run)
2. Run HotRod

### Expected behavior

It should run as usual, without exceptions, and print in logs
```bash
Replacing deprecated Jaeger SDK env var %s with OpenTelemetry env var %s
```

### Relevant log output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa6d5c5]
goroutine 1 [running]:
go.uber.org/zap.(*Logger).clone(...)
 go.uber.org/zap@v1.27.0/logger.go:318
go.uber.org/zap.(*Logger).Sugar(...)
 go.uber.org/zap@v1.27.0/logger.go:147
github.com/jaegertracing/jaeger/internal/jaegerclientenv2otel.MapJaegerToOtelEnvVars(0x0)
 github.com/jaegertracing/jaeger/internal/jaegerclientenv2otel/envvars.go:57 +0x1a5
github.com/jaegertracing/jaeger/examples/hotrod/cmd.onInitialize()
 github.com/jaegertracing/jaeger/examples/hotrod/cmd/root.go:59 +0x29
github.com/spf13/cobra.(*Command).preRun(...)
 github.com/spf13/cobra@v1.8.1/command.go:1019
github.com/spf13/cobra.(*Command).execute(0x1221740, {0x128eb40, 0x0, 0x0})
 github.com/spf13/cobra@v1.8.1/command.go:929 +0x5a2
github.com/spf13/cobra.(*Command).ExecuteC(0x1221180)
 github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
 github.com/spf13/cobra@v1.8.1/command.go:1041
github.com/jaegertracing/jaeger/examples/hotrod/cmd.Execute()
 github.com/jaegertracing/jaeger/examples/hotrod/cmd/root.go:46 +0x26
main.main()
 ./main.go:11 +0xf
```


### Screenshot

_No response_

### Additional context

I'm using the HotRod with the ENV `OTEL_EXPORTER_OTLP_ENDPOINT`. By mistake, I deployed the deprecated Jaeger Agent (as a DaemonSet) with a Kubernetes Service `jaeger-agent` name. So Kubernetes injected the ENV `JAEGER_AGENT_PORT` inside the HotRod pod and the HotRod started crashing with the error.

### Jaeger backend version

v1.58.x

### SDK

_No response_

### Pipeline

_No response_

### Stogage backend

Cassandra

### Operating system

Linux

### Deployment model

Kubernetes, binaries

### Deployment configs

_No response_

--- a/examples/hotrod/cmd/root.go
+++ b/examples/hotrod/cmd/root.go
@@ -56,8 +56,6 @@ func init() {
 
 // onInitialize is called before the command is executed.
 func onInitialize() {
-	jaegerclientenv2otel.MapJaegerToOtelEnvVars(logger)
-
 	zapOptions := []zap.Option{
 		zap.AddStacktrace(zapcore.FatalLevel),
 		zap.AddCallerSkip(1),
@@ -68,6 +66,9 @@ func onInitialize() {
 		)
 	}
 	logger, _ = zap.NewDevelopment(zapOptions...)
+
+	jaegerclientenv2otel.MapJaegerToOtelEnvVars(logger)
+
 	metricsFactory = prometheus.New().Namespace(metrics.NSOptions{Name: "hotrod", Tags: nil})
 
 	if config.MySQLGetDelay != fixDBConnDelay {

